src/metabase/query_processor/middleware/metrics.clj

```diff
@@ -151,7 +151,7 @@
                               1
                               #(merge
                                 %
-                                {:name metric-name}
+                                {:name (lib/column-name query stage-number replacement)}
```

Comment:
This was the incorrect code I mentioned in Slack -- this would get stomped on by the middleware I removed and based on https://github.com/metabase/metabase/pull/48705 this is actually the expected behavior